bundles/org.openhab.binding.lcn/src/main/java/org/openhab/binding/lcn/internal/subhandler/LcnModuleCodeSubHandler.java

```diff
@@ -51,7 +53,7 @@ public void handleRefresh(LcnChannelGroup channelGroup, int number) {
     public void handleStatusMessage(Matcher matcher) {
         String code;
 
-        if (matcher.pattern() == FINGERPRINT_PATTERN) {
+        if (matcher.pattern() == FINGERPRINT_PATTERN_HEX) {
```

Comment:
Is there a reason not checking the decimal representation anymore?

Reply:
IMHO is the "if" case is only for handling the integer parsing. It is only necessary for the hexadecimal representation to handle the regex result differently:

`Integer.parseInt(matcher.group("byte0"), 16)`

Reply:
You're right! The original code could be refactored to be much simpler, though...